apollo-api/src/test/java/com/spotify/apollo/RequestTest.java

```diff
@@ -132,7 +132,32 @@ public void shouldReplaceExistingHeader() throws Exception {
 
     assertThat(requestWithHeader("/foo", "old", "value").withHeaders(newHeaders).headers(),
                is(ImmutableMap.of("old", "value2",
-                                  "newHeader", "value1")));
+                                  "newheader", "value1")));
+  }
+
+  @Test
+  public void shouldTreatHeadersCaseInsensitivelyOnLookup() throws Exception {
+    Request request = Request.forUri("/foo")
+        .withHeader("User-Agent", "value");
+
+    assertThat(request.header("User-Agent"), is(Optional.of("value")));
+  }
+
+  @Test
+  public void shouldTreatHeadersCaseInsensitively() throws Exception {
```

Comment:
I personally don't think header name case is relevant information. The reason it came up specifically is that HTTP/2 mandates lower-case headers: https://http2.github.io/http2-spec/#rfc.section.8.1.2. There's precedent also in how NodeJS treats header casing: https://github.com/nodejs/node-v0.x-archive/issues/1954 - I like the quote about how 'protocol artefacts are not information'.